gitlab-status/handler.go

```diff
@@ -112,3 +135,88 @@ func validateRequest(req []byte) (err error) {
 
 	return nil
 }
+
+func formatLogsURL(gatewayURL string, event *sdk.Event) (string, error) {
+	systemURL, formatErr := formatSystemURL(gatewayURL)
+	if formatErr != nil {
+		return "", fmt.Errorf("error while formatting logs URL: %s", formatErr.Error())
+	}
+
+	return fmt.Sprintf("%s/dashboard/%s/%s/log?repoPath=%s/%s&commitSHA=%s",
+		systemURL, event.Owner, event.Service, event.Owner, event.Repository, event.SHA), nil
+}
+
+func formatEndpointURL(gatewayURL string, event *sdk.Event) (string, error) {
+	systemURL, formatErr := formatSystemURL(gatewayURL)
+	if formatErr != nil {
+		return "", fmt.Errorf("error while formattig endpoint URL: %s", formatErr.Error())
+	}
+	personalURL := strings.Replace(systemURL, SystemSubDomain, event.Owner, -1)
+
+	return fmt.Sprintf("%s/%s", personalURL, event.Service), nil
+}
+
+func formatDashboardURL(gatewayURL string, event *sdk.Event) (string, error) {
+	systemURL, formatErr := formatSystemURL(gatewayURL)
+	if formatErr != nil {
+		return "", fmt.Errorf("error while formatting dashboard URL: %s", formatErr.Error())
+	}
+
+	return fmt.Sprintf("%s/dashboard/%s", systemURL, event.Owner), nil
+}
+
+func formatSystemURL(gatewayURL string) (string, error) {
+	if strings.HasSuffix(gatewayURL, "/") {
+		gatewayURL = strings.TrimSuffix(gatewayURL, "/")
+	}
+	subDomain, err := getSubDomain(gatewayURL)
+	if err != nil {
+		return "", fmt.Errorf("error while geting subdomain for system URL: %s", err)
+	}
+	systemURL := strings.Replace(gatewayURL, subDomain, SystemSubDomain, -1)
+	return systemURL, nil
+}
+
+func getSubDomain(URL string) (string, error) {
```

Comment:
Is this in the Url struct already like `Host` or similar? Check the Go standard library for everything that is available. https://golang.org/pkg/net/url/

Reply:
I have checked that, and the `net/url` does not provide with something like subdomain, since it is subjective (`github.com`)